pkg/collector/corechecks/snmp/checkconfig/config_validate_enrich.go

```diff
@@ -42,45 +60,81 @@ func validateEnrichMetrics(metrics []MetricsConfig) []string {
 			}
 			for i := range metricConfig.MetricTags {
 				metricTag := &metricConfig.MetricTags[i]
-				errors = append(errors, validateEnrichMetricTag(metricTag, metricConfig)...)
+				errors = append(errors, validateEnrichMetricTag(metricTag)...)
 			}
 		}
 	}
 	return errors
 }
 
-func validateEnrichSymbol(symbol *SymbolConfig, metricConfig *MetricsConfig) []string {
+// validateEnrichMetadata will validate MetadataConfig and enrich it.
+func validateEnrichMetadata(metadata MetadataConfig) []string {
+	var errors []string
+	for resName := range metadata {
+		_, isValidRes := validMetadataResources[resName]
+		if !isValidRes {
+			errors = append(errors, fmt.Sprintf("invalid resource: %s", resName))
```

Comment:
why not early returning here?

Reply:
we need to run this code below in all cases:


```
		for i := range metadata[resName].IDTags {
			metricTag := &metadata[resName].IDTags[i]
			errors = append(errors, validateEnrichMetricTag(metricTag)...)
		}
```